### ISSUE TITLE
 🎉 Found a wallet with a non-zero balance! ...but it's every single query iteration. 

### ISSUE THREAD

**Issue by ZombieJesus9000** [cdw1p/evm-crypto-bruteforce#3]
Every single query made is coming back as $0.00 balances but the program is alerting as if it has found a positive balance.

example

[13:42:46] 👾 Address: 0x6DCdF1F7e20d6b9F8c3468B4f07f0139CE82a010
[13:42:46] 💬 Mnemonic: motor talent priority assist orphan dad also earth diet cause page language
[13:42:46] 🔑 Private key: 0x9c42d5966a5f50cf31ae706fba3739f59f90adb852e06f6c2d08fc52457630d1
[13:42:46] 🤑 ETH Balance: $0.00
[13:42:46] 🤑 BNB Balance: $0.00
[13:42:46] 🤑 MATIC Balance:                         
[13:42:46] 🎉 Found a wallet with a non-zero balance!

[13:42:57] 👾 Address: 0x02720910a825dfC5C08f5C3f231B307B9436BD71
[13:42:57] 💬 Mnemonic: indicate until knee ritual valve cry school eager claim inflict dinosaur busy
[13:42:57] 🔑 Private key: 0xea2b20bcece8bd9416b6e54b2986bbd4c4d442bfbbba4e8205f38c1e32fd1547
[13:42:57] 🤑 ETH Balance: $0.00
[13:42:57] 🤑 BNB Balance: $0.00
[13:42:57] 🤑 MATIC Balance:                         
[13:42:57] 🎉 Found a wallet with a non-zero balance!

[13:43:08] 👾 Address: 0x967C34EDD6F833224193ed7Aa136f3BA84fa92C7
[13:43:08] 💬 Mnemonic: diamond rural side animal group exhibit prosper fee casino work arrange tiny
[13:43:08] 🔑 Private key: 0xb0921c71e583e81dd7171dca7c87cdc55de33410d15d4f9443b6ad0fd38be86d
[13:43:08] 🤑 ETH Balance: $0.00
[13:43:08] 🤑 BNB Balance: $0.00
[13:43:08] 🤑 MATIC Balance:                         
[13:43:08] 🎉 Found a wallet with a non-zero balance!

**Comment by irgitau** [comment]
have you seen any other account with a non zero balance?

**Comment by ghost** [comment]
No, I haven't seen any accounts with a non-zero balance. I've seen unexpected data throw off code that didn't handle exceptions though.

**Comment by irgitau** [comment]
Oooh okay... Never thought you'd respond hehe... I've seen a couple of guys
claiming that they got a good bot but they sell... Can't risk to fall into
a scam.. anyways if you find a bot which works well don't forget to hit me
up... Thanks

**Comment by mxdb21** [comment]
same, i've seen many accounts with non zero balance but they're zero actually

**Comment by Frikee** [comment]
Did anybody solve it? There are badly processed outputs in the script. The non-zero balance is actually 0, so it only outputs zero.

**Comment by Lukeronaldo** [comment]
literally the same for me

**Comment by Lukeronaldo** [comment]
matic balance is zero even though they don't show it

**Comment by ZombieJesus9000** [comment]
Well, yeah, the issue with the program is that it is getting some sort of
input that it doesn't know how to parse so after querying the address and
evaluating whatever string it returns for "if balance is equal to zero" is
it coming back false and tripping the non-zero balance logic and then
things go fubar on the rest of the output.

I haven't really dug much into it as this program and the concept behind it
is a flight of fancy novelty that doesn't stand a chance of ever being
successful without some sizable distributed computing behind it. Easy fix
is to just comment out the lines relevant to whichever crypto was causing
the error, I can't recall at the moment.

**Comment by curbmoon** [comment]
You can solve the problem so, in index.js find  and replace

**Chnage the**
function scrapeBlockscan(address, type = 'etherscan') {
  const url = `https://${type}.com/address/${address}`
  return axios.get(url)
    .then(response => {
      const $ = cheerio.load(response.data)
      const balance = $('#ContentPlaceHolder1_divSummary > div.row.g-3.mb-4 > div:nth-child(1) > div > div > div:nth-child(3)').text()
      const balanceResult = balance.split('\n')[4]
      return balanceResult !== undefined ? balanceResult : '$0.00'
    })
    .catch(async () => {
      await delay(10000)
      return '$0.00'
    })
}

**By the**
function scrapeBlockscan(address, type = 'etherscan') {
  const url = `https://${type}.com/address/${address}`
  return axios.get(url)
    .then(response => {
      const $ = cheerio.load(response.data)
      const balance = $('#ContentPlaceHolder1_divSummary > div.row.g-3.mb-4 > div:nth-child(1) > div > div > div:nth-child(3)').text()
      const balancePieces = balance.split('\n');
      if(type === 'polygonscan') {
        const balanceResult = balancePieces[5];
      } else {
        const balanceResult = balancePieces[4];
      }
      return balanceResult !== undefined ? balanceResult : '$0.00'
    })
    .catch(async () => {
      await delay(10000)
      return '$0.00'
    })
}
